terraform/plan.go

```diff
@@ -61,7 +62,12 @@ func (p *Plan) Context(opts *ContextOpts) (*Context, error) {
 		// Even if we're overriding the state, it should be logically equal
 		// to what's in plan. The only valid change to have made by the time
 		// we get here is to have incremented the serial.
-		return nil, errors.New("plan state and ContextOpts state are not equal")
+		//
+		// Due to the fact that serialization may change the representation of
+		// the state, there is little chance that these aren't actually equal.
+		// Log the error condition for reference, but continue with the state
+		// we have.
+		log.Println("[ERROR] plan state and ContextOpts state are not equal")
```

Comment:
Since the serialization process is what's is causing the change, only the raw in-memory representation is going to show any difference. I did try this using spew, but the output was very noise, and we would basically need to create a post-processing tool to diff the pages of output.  I think since the case that's causing this more common than we thought, we should refrain from dumping the entire state into the logs, twice. 

Once we take care of the serialization issues we can make this complain louder.